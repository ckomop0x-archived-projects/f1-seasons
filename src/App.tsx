import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { Footer, Header, Seasons, ErrorBoundary } from './components';
import { Container, themeStyles, NormalizeStyles, GlobalStyles } from './components/styles';

const App: React.FunctionComponent = () => (
  <ThemeProvider theme={themeStyles}>
    <Container id="main-app">
      <NormalizeStyles />
      <GlobalStyles />
      <Header />
      <ErrorBoundary>
        <Seasons />
      </ErrorBoundary>
      <Footer />
    </Container>
  </ThemeProvider>
);

export default App;
