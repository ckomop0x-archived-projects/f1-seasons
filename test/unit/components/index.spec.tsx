import * as React from "react";
import * as renderer from 'react-test-renderer';
import App from "../../../src/scripts/components/App";

it("renders the heading", () => {
    const wrapper = renderer.create(<App/>).toJSON();
    expect(wrapper).toMatchSnapshot()
});