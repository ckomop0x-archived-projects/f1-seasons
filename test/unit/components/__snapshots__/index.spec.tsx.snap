// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the heading 1`] = `
<div
  className="mainContainer"
  id="main-app"
>
  <header
    className="header"
  >
    <a
      className="logo"
      href="/"
      title="Formula 1 winners application"
    >
      <img
        alt="Logo"
        className="logoImage"
        src="test-file-stub"
      />
      <span
        className="logoText"
      >
        F1 standings application
      </span>
    </a>
  </header>
  <div
    className="container"
  >
    <div
      className="seasonsSelector"
    >
      <h1>
        Selected season: 
        2009
      </h1>
      <form
        onChange={[Function]}
      >
        <select
          name="years"
        >
          <option
            value={2009}
          >
            2009
          </option>
          <option
            value={2010}
          >
            2010
          </option>
          <option
            value={2011}
          >
            2011
          </option>
          <option
            value={2012}
          >
            2012
          </option>
          <option
            value={2013}
          >
            2013
          </option>
          <option
            value={2014}
          >
            2014
          </option>
          <option
            value={2015}
          >
            2015
          </option>
          <option
            value={2016}
          >
            2016
          </option>
          <option
            value={2017}
          >
            2017
          </option>
          <option
            value={2018}
          >
            2018
          </option>
        </select>
      </form>
    </div>
    <div
      className="seasons"
    />
  </div>
  <footer
    className="footer"
  >
    Â© 2018 Pavel Klochkov
  </footer>
</div>
`;
